CMakeLists.txt

```diff
@@ -49,8 +49,6 @@ endif()
 
 if(WITH_STL)
   # Require at least C++17. C++20 is needed to avoid gsl::span
-  add_definitions(-DHAVE_CPP_STDLIB -DHAVE_GSL)
```

Comment:
I have minor doubt here. Don't we need this definition in the main common place for other projects that may be including API headers directly, i.e. if I build a test app that consumes API, do I need to add a dependency on `opentelemetry_api` target everywhere? If another target is created, such as that would aggregate multiple targets into one, would that new target also have to add those two definitions?

Trying to understand the _cons_ of having this possibly defined in both spots, main `CMakeLists.txt`, as well as in the api?

Reply:
You mean we should just add `HAVE_CPP_STDLIB` and `HAVE_GSL` into `INTERFACE_COMPILE_DEFINITIONS` of `opentelemetry_resources` in `opentelemetry_sdk`? It seems better, I will test it later.